summer_staj.py

```diff
@@ -25,26 +31,30 @@
 ########
 
 # some global variables to create two csv files on .txt format
-first_outfile = "data/first.txt"
-second_outfile = "data/second.txt"
+first_outfile = "data/first.csv"
+second_outfile = "data/second.csv"
 second_infile = first_outfile
 
+CUSTOM_SETTINGS = { 'DOWNLOAD_DELAY': 0.1,
+                    # 'ITEM_PIPELINES': {'freedom.pipelines.IndexPipeline': 300 }
+                    }
+
 """
     First of the spiders. Just goes through the first sets of pages and extracts links 
 with some other information.
 """
 
 class first_spider(scrapy.Spider):
     name = 'first_spider'
+    custom_settings = CUSTOM_SETTINGS
+    def __init__(self, *args, **kwargs):
+        super(first_spider, self).__init__(*args, **kwargs)
+        self.companies = dict()

```

Comment:
a single spider's crawlers can contribute to a single dictionary which can later be used to write to a file instead of tiring the crawlers with unordered writes to a single file.